src/chunk.c

```diff
@@ -1099,21 +1099,47 @@ chunk_create_from_hypercube_and_table_after_lock(const Hypertable *ht, Hypercube
 	chunk->hypertable_relid = ht->main_table_relid;
 	Assert(OidIsValid(ht->main_table_relid));
 
+	Relation chunk_rel = table_open(chunk_table_relid, AccessExclusiveLock);
+	Relation ht_rel = table_open(ht->main_table_relid, AccessExclusiveLock);
```

Comment:
Since we are only reading from this hypertable, AccessShareLock should be enough. We don't want to block other backends unnecessarily.

```suggestion
	Relation ht_rel = table_open(ht->main_table_relid, AccessShareLock);
```

Reply:
This method is called "_after_lock", the hypertable is locked in the caller.

Reply:
It makes sense to get a lock strong enough to block table modifications even if it is locked in the caller. Otherwise, bugs can creep in that create race conditions.